pkg/sql/rename_table.go

```diff
@@ -105,6 +107,13 @@ func (n *renameTableNode) startExec(params runParams) error {
 		return err
 	}
 
+	if newTn.ExplicitSchema && newTn.Schema() == tree.PublicSchema && tableDesc.Temporary {
```

Comment:
We are going to add user defined schemas, so let's be proactive and check if the new schema isn't the temporary schema.